ENH Add libsvm-like calibration procedure to CalibratedClassifierCV (#17856)
scikit-learn/scikit-learn: doc/whats_new/v0.24.rst

  sparse matrix or dataframe at the start. :pr:`17546` by
  :user:`Lucy Liu <lucyleeow>`.

- |Enhancement| Add `ensemble` parameter to
  :class:`calibration.CalibratedClassifierCV`, which enables implementation
  of calibration via an ensemble of calibrators (current method) or
  just one calibrator using all the data (similar to the built-in feature of
  :mod:`sklearn.svm` estimators with the `probabilities=True` parameter).

>   :pr:`17856` by :user:`Lucy Liu <lucyleeow>`.

Reviewer: I think we should also credit @aesuli for the original issue and PR

(sorry, I don't have time to make another real review at the moment)

Author: Thanks I forgot!

  sparse matrix or dataframe at the start. :pr:`17546` by
  :user:`Lucy Liu <lucyleeow>`.

- |Enhancement| Refactor :class:`calibration.CalibratedClassifierCV` so the
  chaining of the classifier to its corresponding calibrator and fitting of

>   the calibrator is now separate. Also add `ensemble` parameter, which

Reviewer 2: I think you can skip the mention of the internal refactoring and focus on the new `ensemble` parameters which is what is really of interest to our users.

  implements calibration via an ensemble of calibrators (current method) or

>   just one calibrator using all the data (method implemented in

Reviewer 2: ```suggestion
  just one calibrator using all the data (similar to the built-in feature implemented in
```

>   just one calibrator using all the data (method implemented in
>   :mod:`sklearn.svm` estimators with the `probabilities=True` parameter).
>   :pr:`17856` by :user:`Lucy Liu <lucyleeow>`.

Reviewer 2: ```suggestion
  just one calibrator using all the data (similar to the built-in feature of
  :mod:`sklearn.svm` estimators with the `probabilities=True` parameter).
  :pr:`17856` by :user:`Lucy Liu <lucyleeow>`.
```
